smqtk_classifier/interfaces/classify_image_supervised.py

```diff
@@ -0,0 +1,79 @@
+import abc
+from typing import Any, Iterable, Mapping
+from PIL import Image
+
+from .classify_image import ClassifyImage
+
+
+class ClassifyImageSupervised(ClassifyImage):
+    """
+    Class of classifiers that are trainable via supervised training, i.e. are
+    given specific Image examples for class labels.
+    """
+
+    @abc.abstractmethod
+    def has_model(self) -> bool:
+        """
+        :return: If this instance currently has a model loaded. If no model is
+            present, classification of Images cannot happen (needs to be
+            trained).
+        :rtype: bool
+        """
+
+    def train(
+        self,
+        class_examples: Mapping[Any, Iterable[Image.Image]],
```

Comment:
I assume `Any` here should possibly be `Hashable`, or if that is an issue, a more specific `Union` of common label types (e.g. `Union[int, str]`).